plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -57,42 +57,56 @@ export class KubernetesBuilder {
 
   public async build() {
     const logger = this.env.logger;
+    const config = this.env.config;
 
     logger.info('Initializing Kubernetes backend');
 
-    const customResources = this.buildCustomResources();
+    if (config.has('kubernetes')) {
```

Comment:
Here's a much more coarse escape hatch for avoiding building things when the config is missing.

The problem is we can't just return `{}` at the bottom, because the router ends up being undefined and the backend fails to initialize, logs like:

```
2022-02-14T22:04:54.709Z kubernetes warn Failed to initialize kubernetes backend: kubernetes config is missing type=plugin
Backend failed to start up Error: Expected a callback function but got a [object Undefined]
```